### PR TITLE
osutil/mount: de-duplicate code to use a list

### PR DESCRIPTION
**Pull request by anonymouse64**
This is slightly less performant due to the loop and variables, but is more readable and less susceptible to typos and the ilk that plague repetitive code.

For clarity, the difference in performance is:
```
benchmark                                    old ns/op     new ns/op     delta
BenchmarkUnmountFlagsToOptsAllMissing-32     2.90          6.20          +113.79%
BenchmarkUnmountFlagsToOptsMixed-32          48.1          60.6          +25.99%
BenchmarkUnmountFlagsToOptsAllPresent-32     262           285           +8.78%
BenchmarkMountFlagsToOptsAllMissing-32       9.89          9.20          -6.98%
BenchmarkMountFlagsToOptsAllPresent-32       406           474           +16.75%
BenchmarkMountFlagsToOptsMixed-32            66.2          67.0          +1.21%

benchmark                                    old allocs     new allocs     delta
BenchmarkUnmountFlagsToOptsAllMissing-32     0              0              +0.00%
BenchmarkUnmountFlagsToOptsMixed-32          1              1              +0.00%
BenchmarkUnmountFlagsToOptsAllPresent-32     3              3              +0.00%
BenchmarkMountFlagsToOptsAllMissing-32       0              0              +0.00%
BenchmarkMountFlagsToOptsAllPresent-32       4              4              +0.00%
BenchmarkMountFlagsToOptsMixed-32            1              1              +0.00%

benchmark                                    old bytes     new bytes     delta
BenchmarkUnmountFlagsToOptsAllMissing-32     0             0             +0.00%
BenchmarkUnmountFlagsToOptsMixed-32          16            16            +0.00%
BenchmarkUnmountFlagsToOptsAllPresent-32     112           112           +0.00%
BenchmarkMountFlagsToOptsAllMissing-32       0             0             +0.00%
BenchmarkMountFlagsToOptsAllPresent-32       240           240           +0.00%
BenchmarkMountFlagsToOptsMixed-32            16            16            +0.00%
```

This is with go 1.13 and calculated using golang.org/x/tools/cmd/benchcmp

### CODE REVIEW

File: osutil/mount/mount_linux.go [hunk 1]
```diff
@@ -23,61 +23,51 @@ import (
...
+}
+
+func flagOptSearch(flags int, lookupTable []syscallNumPair) (opts []string, unknown int) {
+	var f, i int
+	var sys syscallNumPair
+	for i, sys = range lookupTable {
+		f = sys.num
+		if flags&f == f {
+			flags ^= f
+			opts = append(opts, lookupTable[i].str)
```
Comment (zyga) [inline osutil/mount/mount_linux.go]:
```suggestion
			opts = append(opts, sys.str)
```

File: osutil/mount/mount_linux.go [hunk 2]
```diff
@@ -23,61 +23,51 @@ import (
...
+	{"UMOUNT_NOFOLLOW", UMOUNT_NOFOLLOW},
+	{"MNT_FORCE", syscall.MNT_FORCE},
+	{"MNT_DETACH", syscall.MNT_DETACH},
+	{"MNT_EXPIRE", syscall.MNT_EXPIRE},
+}
+
+func flagOptSearch(flags int, lookupTable []syscallNumPair) (opts []string, unknown int) {
+	var f, i int
+	var sys syscallNumPair
+	for i, sys = range lookupTable {
```
Comment (zyga) [inline osutil/mount/mount_linux.go]:
```suggestion
	for _, sys := range lookupTable {
```

File: osutil/mount/mount_linux.go [hunk 3]
```diff
@@ -23,61 +23,51 @@ import (
...
+var unmountSyscalls = []syscallNumPair{
+	{"UMOUNT_NOFOLLOW", UMOUNT_NOFOLLOW},
+	{"MNT_FORCE", syscall.MNT_FORCE},
+	{"MNT_DETACH", syscall.MNT_DETACH},
+	{"MNT_EXPIRE", syscall.MNT_EXPIRE},
+}
+
+func flagOptSearch(flags int, lookupTable []syscallNumPair) (opts []string, unknown int) {
+	var f, i int
+	var sys syscallNumPair
```
Comment (bboozzoo) [inline osutil/mount/mount_linux.go]: Extra points:
```
import "math/bits"

var allKnown int = 0xffffffff
func init() {
    allKnown = 0
    for _, lookup := range [][]syscallNumPair{mountSyscalls, unmountSyscalls} {
        for _, en := range lookup {
            allKnown |= en.num
        }
    }
}
...
func flagOptSearch(flags int, lookupTable []syscallNumPair) (opts []string, unknown int) {
    known := flags & allKnown
    if known = 0 {
        return nil, flags & ~allKnown
    }
    // pre-allocate based on the popcount of known flags
    opts = make([]string, 0, bits.OnesCount(known))
    ```
}
```

Comment (zyga) [reply osutil/mount/mount_linux.go]: I've optimised this a bit actually :) coming up after this lands.

**Pull request merged**